keras/utils/generic_utils.py

```diff
@@ -266,7 +266,8 @@ def __init__(self, target, width=30, verbose=1, interval=0.05):
         self.interval = interval
         self.total_width = 0
         self.seen_so_far = 0
-        self.verbose = verbose
+        self.verbose = verbose 
+        self.is_jupyter = 'ipykernel' in sys.modules
```

Comment:
Please prefer e.g. `self._dynamic_display = sys.stdout.isatty() or 'ipykernel' in sys.modules` (private attribute, single flag, not jupyter specific)